lib/SILOptimizer/Analysis/ClassHierarchyAnalysis.cpp

```diff
@@ -32,7 +32,7 @@ class NominalTypeWalker: public ASTWalker {
 
   bool walkToDeclPre(Decl *D) override {
     auto *NTD = dyn_cast<NominalTypeDecl>(D);
-    if (!NTD)
+    if (!NTD || !NTD->hasInterfaceType())
```

Comment:
By the time the SIL optimizer runs, the type checker has been torn down so nothing "new" can get type checked that wasn't type checked before.